R/kernel.r

```diff
@@ -312,6 +316,8 @@ initialize = function(connection_file) {
 
     executor <<- Executor$new(send_response = .self$send_response,
         abort_queued_messages = .self$abort_queued_messages)
+    comm_manager <<- Comm_Manager$new(send_response = .self$send_response)
+    assign('kernel', .self, envir = .GlobalEnv)
```

Comment:
also if the only use of the kernel object is to provide an interface to adding and removing comm managers, we should instead expose an object that allows us to do only that.

i suggest

1. documenting `CommManager` and `Comm` by `#'@export` before the class and adding docstrings to methods
2. only making the `comm_manager` available, not the kernel